tests/unit/test_directory_conditions.py

```diff
@@ -110,12 +110,12 @@ def test_disallowed_directory_conditions(
 
     try:
         with pytest.raises(expected_exception=ValueError, match=expected_exception):
-            dataset_converter = nwb2bids.DatasetConverter.from_nwb_directory(
-                nwb_directory=minimal_nwbfile_path.parent, additional_metadata_file_path=additional_metadata_file_path
+            nwb_paths = [minimal_nwbfile_path]
+            dataset_converter = nwb2bids.DatasetConverter.from_nwb_paths(
+                nwb_paths=nwb_paths, additional_metadata_file_path=additional_metadata_file_path
             )
             dataset_converter.extract_metadata()
 
             dataset_converter.write_dataset_description(bids_directory=bids_directory)
     finally:
-        if "implicit" in test_case:
-            os.chdir(initial_working_directory)
+        os.chdir(initial_working_directory)
```

Comment:
See the other comment about using monkeypatch

Reply:
Swapped